### DISCUSSION TITLE
dmrfill - Fill codeplugs with repeater data

### DISCUSSION THREAD

**Discussion by jancona** [hmatuschek/qdmr#471] [Show and tell]
Hi all,
[https://github.com/jancona/dmrfill] is a tool to simplify the process of adding repeaters and talkgroups to codeplugs. When I first got my DMR radio, I found the whole process around codeplus very frustrating. Even if your local area repeater group maintains a codeplug, as soon as you travel you're back to trying to find a codeplug with the area repeaters or at least locate repeater and talkgroup data so you can build your own.

`dmrfill` trys to solve this problem by gathering repeater data and adding it an existing codeplug. So, you start with a minimal codeplug in [QDMR Extensible Codeplug](https://dm3mat.darc.de/qdmr/manual/ch03.html) format and use dmrfill to query [RadioID](https://radioid.net/) or [RepeaterBook](https://www.repeaterbook.com/) for repeater data and then create the proper zones, channels, contacts, and group lists. You can do this multiple times for different areas by piping the results through dmrfill multiple times, building up a custom codeplug.

See the [README](https://github.com/jancona/dmrfill/blob/master/README.md) for more details. You can download a version for your architecture [here](https://github.com/jancona/dmrfill/releases). The [0.2.0 release](https://github.com/jancona/dmrfill/releases/tag/v0.2.0) works with QDMR/dmrconf 0.12.0. The older releases work with QDMR 0.11.x.

I'm happy to answer questions and I'd love to hear any questions, suggestions or issues you might have.

Jim
N1ADJ

**Comment by allesand** [comment]
Very cool! Since this question comes up quite often when hams try out DMR, they now can use a modern CPS *and* have the codeplug building almost automated.

**Comment by ignBiometrical** [comment]
Hey, I love the look of this project. I have attempted to use QDMR, and dmrfill, however attempting to use a generated yaml gives me an error. I attempted to use a custom command, as well as the given example one, and then get the following error when importing into QDMR:

```
Debug in src/application.cc@347: Load codeplug from '/home/jacob/Downloads/maine.codeplug.yaml'.
Debug in lib/config.cc@372: Using format version 0.11.2.
terminate called after throwing an instance of 'YAML::TypedBadConversion<double>'
  what():  yaml-cpp: error at line 351, column 20: bad conversion
Aborted
```

**Reply by hmatuschek** [reply]
You may need to create an issue over at 

https://github.com/jancona/dmrfill/issues

as dmrfill is developed there.

**Reply by jancona** [reply]
Where did you get the `maine.codeplug.yaml` file? It looks like it was produced by an old version of dmrfill. As @hmatuschek said, please file an issue on the dmrfill repo. I'll try to help you there.
